src/OpenTelemetry.Extensions.Hosting/Implementation/TracerProviderBuilderHosting.cs

```diff
@@ -83,6 +84,12 @@ public TracerProviderBuilder Configure(Action<IServiceProvider, TracerProviderBu
 
         public TracerProvider Build(IServiceProvider serviceProvider)
         {
+            var configActions = serviceProvider.GetRequiredService<IEnumerable<IConfigureOptions<TracerProviderBuilder>>>();
```

Comment:
This is either totally brilliant or misuse of the options API 😄 I suppose `TracerProviderBuilder` is kind of an options class? I think the intention with `IConfigureOptions` is the registrations are invoked on options classes created when fed into something via IOptions<T> (or its cousins). This is kind of co-opting that mechanism for our own purpose, which isn't necessarily a bad thing. We could have our own `IConfigureTracerProviderBuilder` that does essentially the same thing without the co-opt.